examples/fluid.py

```diff
@@ -0,0 +1,233 @@
+# http://developer.download.nvidia.com/books/HTML/gpugems/gpugems_ch38.html
+
+import taichi as ti
+import numpy as np
+
+res = 600
+half_res = res * 0.5
+dx = 1.0
+rdx = 1.0 / dx
+half_rdx = 0.5 * rdx
+dt = 0.025
+v_decay = 0.95
+p_jacobi_iters = 30
+f_strength = 10000.0
+
+enable_diffusion = False
+viscority = 10.0
+v_jacobi_iters = 20
+
+ti.init(arch=ti.cuda)
+
+velocities = ti.Vector(2, dt=ti.f32, shape=(res, res))
+new_velocities = ti.Vector(2, dt=ti.f32, shape=(res, res))
+div_vels = ti.var(dt=ti.f32, shape=(res, res))
+pressures = ti.var(dt=ti.f32, shape=(res, res))
+new_pressures = ti.var(dt=ti.f32, shape=(res, res))
+color_buffer = ti.Vector(3, dt=ti.f32, shape=(res, res))
+
+
+@ti.func
+def in_boundary(i, j):
+    return 0 <= i and i < res and 0 <= j and j < res
```

Comment:
`Taichi` supports `0 <= i < res`

Reply:
I didn't even know this is valid python syntax...